packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_error.dart

```diff
@@ -44,6 +44,37 @@ final class UnsupportedUserLocation implements VertexAIException {
   String get message => _message;
 }
 
+/// Exception thrown when the service api is not enabled.
+final class ServiceApiNotEnabled implements VertexAIException {
+  /// Constructor
+  ServiceApiNotEnabled(String projectId)
+      : _message = _messageWithProjectId(projectId);
+
+  final String _message;
+
+  static String _messageWithProjectId(String projectId) =>
+      'The Vertex AI in Firebase SDK requires the Vertex AI in Firebase API '
+      '(`firebasevertexai.googleapis.com`) to be enabled in your Firebase project. Enable this API '
+      'by visiting the Firebase Console at '
+      'https://console.firebase.google.com/$projectId/genai '
+      'and clicking "Get started". If you enabled this API recently, wait a few minutes for the '
+      'action to propagate to our systems and then retry.';
+
+  @override
+  String get message => _message;
+}
```

Comment:
Consider storing the constructor argument and creating the message lazily.

```suggestion
  ServiceApiNotEnabled(this._projectId);

  final String _projectId;

  @override
  String get message =>
      'The Vertex AI in Firebase SDK requires the Vertex AI in Firebase API '
      '(`firebasevertexai.googleapis.com`) to be enabled in your Firebase project. Enable this API '
      'by visiting the Firebase Console at '
      'https://console.firebase.google.com/$_projectId/genai '
      'and clicking "Get started". If you enabled this API recently, wait a few minutes for the '
      'action to propagate to our systems and then retry.';
}
```